core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts

```diff
@@ -704,6 +741,284 @@ export class LoggedInUser extends BaseUser {
     }
   }
 
+  /**
+   * Function for navigating to the creator dashboard page.
+   */
+  async navigateToCreatorDashboardPage(): Promise<void> {
+    await this.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * Function for navigating to the learner dashboard page.
+   */
+  async navigateToLearnerDashboardPage(): Promise<void> {
+    await this.goto(learnerDashboardUrl);
+  }
+
+  /**
+   * Function for navigating to the learner dashboard page.
+   */
+  async openCommunityLessonsTab(): Promise<void> {
+    if (this.page.url() !== learnerDashboardUrl) {
+      await this.goto(learnerDashboardUrl);
+    }
+    await this.page.click(communityLessonsTabButton);
+    await this.page.waitForSelector('h1.lessons-title');
+  }
+
+  /**
+   * Function for navigating to the creator dashboard page.
+   */
+  async navigateToCommunityLibraryPage(): Promise<void> {
+    await this.goto(communityLibraryUrl);
+  }
+
+  /**
+   * Function for opening an exploration from the community library page.
+   */
+  async searchAndOpenExplorationFromCommunityLibrary(
+    explorationName: string
+  ): Promise<void> {
+    if (this.page.url() !== communityLibraryUrl) {
+      await this.navigateToCommunityLibraryPage();
+    }
+    await this.findExplorationInCommunityLibrary(explorationName);
+    await this.page.click(communityExplorationCard);
+    await this.page.waitForNavigation();
+  }
+
+  /**
+   * This function checks whether we have completed an exploration or not.
+   */
+  async expectReachedTheEndOfExploration(): Promise<void> {
+    if (await this.page.waitForSelector(explorationCompletionToast)) {
+      showMessage('Exploration completed successfully!');
+    } else {
+      throw new Error('Failed to complete the exploration.');
+    }
+  }
+
+  /**
+   * Function for reporting an exploration.
+   */
+  async reportTheExploration(reportMessage: string): Promise<void> {
+    await this.page.click(reportExplorationButton);
+    await this.page.waitForSelector('.modal-content');
+    await this.page.click(reportOtherOptionSelector);
+    await this.page.waitForSelector(reportTextArea);
+    await this.type(reportTextArea, reportMessage);
+    await this.page.click('.e2e-test-submit-report-button');
+    await this.page.waitForSelector('.modal-content');
+    await this.page.click(
+      'oppia-exploration-successfully-flagged-modal button'
+    );
+    /** Giving explicit timeout because we need to wait for small
+     * transition to complete. We cannot wait for the next element to click
+     * using its selector as it is instantly loaded in the DOM but cannot
+     * be clicked until the transition is completed.
+     */
+    await this.page.waitForTimeout(500);
+    showMessage(`Exploration reported with message ${reportMessage}.`);
+  }
+
+  /**
+   * Function for giving feedback for an exploration.
+   */
+  async giveFeedbackToExploration(feedbackMessage: string): Promise<void> {
+    await this.page.click(explorationFeedbackButton);
```

Comment:
Try waiting for the `explorationFeedbackButton` selector before clicking on it (and ensure it is enabled if necessary).

Reply:
Thanks, updated it.